cmd/print.go

```diff
@@ -73,39 +72,103 @@ func (pr *PrintRunner) PrintGatewaysAndHTTPRoutes(cmd *cobra.Command, _ []string
 		return fmt.Errorf("failed to initialize namespace filter: %w", err)
 	}
 
-	httpRoutes, gateways, err := i2gw.ToGatewayAPIResources(cmd.Context(), pr.namespaceFilter, pr.inputFile, pr.providers)
+	gatewayResources, err := i2gw.ToGatewayAPIResources(cmd.Context(), pr.namespaceFilter, pr.inputFile, pr.providers)
 	if err != nil {
 		return err
 	}
 
-	pr.outputResult(httpRoutes, gateways)
+	pr.outputResult(gatewayResources)
 
 	return nil
 }
 
-func (pr *PrintRunner) outputResult(httpRoutes []gatewayv1.HTTPRoute, gateways []gatewayv1.Gateway) {
-	if len(httpRoutes)+len(gateways) == 0 {
-		msg := "No resources found"
-		if pr.namespaceFilter != "" {
-			msg = fmt.Sprintf("%s in %s namespace", msg, pr.namespaceFilter)
+func (pr *PrintRunner) outputResult(gatewayResources []i2gw.GatewayResources) {
+	resourceCount := 0
+
+	for _, r := range gatewayResources {
+		resourceCount += len(r.GatewayClasses)
+		for _, gatewayClass := range r.GatewayClasses {
+			gatewayClass := gatewayClass
+			err := pr.resourcePrinter.PrintObj(&gatewayClass, os.Stdout)
+			if err != nil {
+				fmt.Printf("# Error printing %s GatewayClass: %v\n", gatewayClass.Name, err)
+			}
+		}
+	}
```

Comment:
It's just a nitpick, but is there any reason you repeated the outer for loop instead of having it once?

Reply:
Kept the original logic: print all gateways from all providers, then all HTTPRoutes etc.

If it's in one loop, then the logic would change and we'd print all resources per provider.

Reply:
100%